modules/decline/test/src/smithy4s/decline/OptsSchematicSpec.scala

```diff
@@ -34,41 +32,35 @@ import smithy4s.schema.Schema._
 import weaver._
 
 object OptsSchematicSpec extends SimpleIOSuite {
-  type EitherThrow[A] = Either[Throwable, A]
-  private val compiler = new OptsVisitor[EitherThrow]
-
   def sampleStruct[A](name: String, schema: Schema[A]): Schema[A] =
     struct(
       schema.required[A](name, identity)
     )(identity)
 
-  def parseOpts[A](
-                    schema: Schema[A]
-                  )(
-                    input: String*
-                  ): Either[Help, EitherThrow[A]] = Command("test-command", "test-header")(
-    schema.compile[OptsVisitor.OptsF[EitherThrow, *]](compiler)
-  )
-    .parse(input)
+  def parseOpts[A](schema: Schema[A])(input: String*): Either[Help, A] =
+    Command("test-command", "test-header")(schema.compile[Opts](OptsVisitor)).parse(input) match {
```

Comment:
you probably want to drop the following

also, I learnt about that not too long ago: https://www.scala-lang.org/api/2.13.x/scala/util/ChainingOps.html